static/src/javascripts/test/spec/common/commercial/third-party-tags/outbrain.spec.js

```diff
@@ -1,8 +1,283 @@
+
+            sut.init();
+            mediator.emit('modules:commercial:dfp:rendered', eventStub);
+            expect(sut.load).not.toHaveBeenCalled();
+        });
+
+        it('should not load when user is logged in', function () {
+            identity.isUserLoggedIn = function () {
+                return true;
+            };
+            spyOn(sut, 'load');
+
+            sut.init();
+            mediator.emit('modules:commercial:dfp:rendered', eventStub);
+            expect(sut.load).not.toHaveBeenCalled();
+        });
+
+        it('should not load when isPreview', function () {
+            config.page.isPreview = true;
+            spyOn(sut, 'load');
+
+            sut.init();
+            mediator.emit('modules:commercial:dfp:rendered', eventStub);
+            expect(sut.load).not.toHaveBeenCalled();
+        });
+
+        it('should not load when merchandising-high component is not empty', function () {
+            spyOn(sut, 'load');
+
+            eventStub.isEmpty = false;
+
+            sut.init();
+            mediator.emit('modules:commercial:dfp:rendered', eventStub);
+
+            expect(sut.load).not.toHaveBeenCalled();
+        });
+    });
+
+    describe('Sections', function () {
+        it('should return "sections" for news sections', function () {
+            config.page.section = 'uk-news';
+            expect(sut.getSection()).toEqual('sections');
+
+            config.page.section = 'us-news';
+            expect(sut.getSection()).toEqual('sections');
+
+            config.page.section = 'au-news';
+            expect(sut.getSection()).toEqual('sections');
+        });
+
+        it('should return "sections" for selected sections', function () {
+            config.page.section = 'politics';
+            expect(sut.getSection()).toEqual('sections');
+
+            config.page.section = 'world';
+            expect(sut.getSection()).toEqual('sections');
+
+            config.page.section = 'business';
+            expect(sut.getSection()).toEqual('sections');
+
+            config.page.section = 'commentisfree';
+            expect(sut.getSection()).toEqual('sections');
+        });
+
+        it('should return "all" for all other sections', function () {
+            config.page.section = 'culture';
+            expect(sut.getSection()).toEqual('all');
+
+            config.page.section = 'football';
+            expect(sut.getSection()).toEqual('all');
+        });
+    });
+
+    describe('Load', function () {
+        it('should create two containers for desktop with correct IDs for "sections"', function (done) {
+            detect.getBreakpoint = function () {
+                return 'desktop';
+            };
+            sut.getSection = function () {
+                return 'sections';
+            };
+
+            sut.load();
+
+            fastdom.defer(function () {
+                expect($('.OUTBRAIN', $fixtureContainer).first().data('widgetId')).toEqual('AR_12');
```

Comment:
If an expectation fails within a fastdom callback, does it get reported to the test runner?